packages/api/test/pin.spec.js

```diff
@@ -4,256 +4,409 @@ import { endpoint } from './scripts/constants.js'
+          cid: 'bafybeibqmrg5e5bwhx2ny4kfcjx2mm3ohh2cd4i54wlygquwx7zbgwqs4e',
+          meta: 1
+        })
       })
+
+      assert(res, 'Server responded')
+      assert.strictEqual(res.status, ERROR_CODE)
+      const data = await res.json()
+      const error = data.error
+      assert.strictEqual(error.reason, ERROR_STATUS)
+      assert.strictEqual(error.details, INVALID_META)
     })
 
-    assert(res, 'Server responded')
-    assert.strictEqual(res.status, ERROR_CODE)
-    const data = await res.json()
-    const error = data.error
-    assert.strictEqual(error.reason, ERROR_STATUS)
-    assert.strictEqual(error.details, INVALID_NAME)
+    it('validates meta values', async () => {
+      const res = await fetch(new URL('pins', endpoint).toString(), {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${token}`,
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          cid: 'bafybeibqmrg5e5bwhx2ny4kfcjx2mm3ohh2cd4i54wlygquwx7zbgwqs4e',
+          meta: {
+            app_id: 1
+          }
+        })
+      })
+
+      assert(res, 'Server responded')
+      assert.strictEqual(res.status, ERROR_CODE)
+      const data = await res.json()
+      const error = data.error
+      assert.strictEqual(error.reason, ERROR_STATUS)
+      assert.strictEqual(error.details, INVALID_META)
+    })
   })
 
-  it('validates origins', async () => {
-    const res = await fetch(new URL('pins', endpoint).toString(), {
-      method: 'POST',
-      headers: {
-        Authorization: `Bearer ${token}`,
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({
-        cid: 'bafybeibqmrg5e5bwhx2ny4kfcjx2mm3ohh2cd4i54wlygquwx7zbgwqs4e',
-        origins: 1
-      })
+  describe('GET /pins/:requestId', () => {
+    let token = null
+    before(async () => {
+    // Create token
+      token = await getTestJWT()
     })
 
-    assert(res, 'Server responded')
-    assert.strictEqual(res.status, ERROR_CODE)
-    const data = await res.json()
-    const error = data.error
-    assert.strictEqual(error.reason, ERROR_STATUS)
-    assert.strictEqual(error.details, INVALID_ORIGINS)
-  })
+    it('returns anauthorized if no token', async () => {
+      const res = await fetch(new URL('pins/1', endpoint).toString(), {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      })
+
+      assert(res, 'Server responded')
+      assert(!res.ok)
+      assert.deepEqual(res.status, 401)
+    })
 
-  it('validates meta', async () => {
-    const res = await fetch(new URL('pins', endpoint).toString(), {
-      method: 'POST',
-      headers: {
-        Authorization: `Bearer ${token}`,
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({
-        cid: 'bafybeibqmrg5e5bwhx2ny4kfcjx2mm3ohh2cd4i54wlygquwx7zbgwqs4e',
-        meta: 1
+    it('requires a string as requestId', async () => {
```

Comment:
nit: I think we can improve this to test whether requestId is a string that can be coerced into number